templates/common/config/ironic.conf

```diff
@@ -54,6 +58,15 @@ user_domain_name=Default
 project_name=service
 project_domain_name=Default
 
+[service_user]
```

Comment:
I had to go back and figure out exactly what we did. The tl;dr is you don't need the separate service_user configuration section like in nova/cinder. The only thing required is the service_token_roles_required portion above.

We didn't end up creating an extra user and applying the extra configuration under the guise of an extra service user, because we already had a service project/scoped user for talking to cinder, and cinder's requirement was a service scoped user *and* the token request for validation.

As a result, we taught ironic when to request authentication with a service token bundled with the user token from the request, which worked and unblocked CI allowing jobs to again pass.

Reply:
oh ok.
so ye do not actully wrap the auth plugin like this
https://github.com/openstack/nova/blob/master/nova/service_auth.py#L33-L55

and done need a service_user group as a result in ironic.

to be clear we don't actuull require a separate service user
each of the services should have a user in keystone and the common code for that user creation give that user both the admin and service roles.  so we just use nova  and ironic would work fine for you.

but i guess ye are use the cinder section to generate it when talking to ciner ectra.

ok then ill drop this and the init contiaenr change 
thanks for the review

Reply:
Yup, Exactly! Thanks!